lib/jsdom/level2/html.js

```diff
@@ -2110,6 +2139,27 @@ define('HTMLFrameElement', {
         }
       }
     },
+
+    _detach: function () {
+      core.HTMLElement.prototype._detach.call(this);
+
+      var self = this;
+      var parent = this._ownerDocument.defaultView;
+
+      if (self.contentWindow) {
+        self.contentWindow.close();
+      }
+
+      // delete frame reference
+      if (self.getAttribute("name")) {
+        parent._frame(self.getAttribute("name"));
+      }
+
+      parent._length--;
```

Comment:
Can probably also move this to logic (including deletion) to refreshAccessors. Delete all accessors from 0 to `window._length`, then set `window._length` to `frames.length`, then add new accessors.